lib/livebook/application.ex

```diff
@@ -83,6 +84,22 @@ defmodule Livebook.Application do
     end
   end
 
+  # See https://github.com/elixir-nx/livebook/issues/302
+  defp validate_hostname_resolution!() do
+    unless Livebook.Config.longname() do
+      hostname = Livebook.Utils.node_host() |> to_charlist()
+
+      if :inet.gethostbyname(hostname) == {:error, :nxdomain} do
+        raise RuntimeError, """
```

Comment:
Can you check if this error message is readable locally? Often errors from application start is not readable at all, so we tend to use Logger.error and then call System.halt(1).

Reply:
We already have `Livebook.Config.abort!`, forgot to use that :)

Reply:
![image](https://user-images.githubusercontent.com/17034772/119978503-01cd8300-bfba-11eb-8257-0191b4b7bfc2.png)